back-end/src/sciler/config/checkConfig.go

```diff
@@ -0,0 +1,367 @@
+				valueType.String(), value)
+		}
+	case "boolean":
+		if valueType != reflect.Bool {
+			return fmt.Errorf("instruction type boolean expected but %s found as type of value %v",
+				valueType.String(), value)
+		}
+	case "numeric":
+		if valueType != reflect.Int && valueType != reflect.Float64 {
+			return fmt.Errorf("instruction type numeric expected but %s found as type of value %v",
+				valueType.String(), value)
+		}
+	case "array":
+		if valueType != reflect.Slice {
+			return fmt.Errorf("instruction type array/slice expected but %s found as type of value %v",
+				valueType.String(), value)
+		}
+	default:
+		return fmt.Errorf("custom types like: %s, are not yet implemented", instructionType)
+	}
+	return nil
+}
+
+// checkActionLabel checks if there is a label with this ID,
+// and checks if all components under this label have the correct instructions with a call to checkActionDevice
+// if the config does not follow the manual, a non-empty list of mistakes is returned
+func checkActionLabel(action Action, config WorkingConfig) []string {
+	errorList := make([]string, 0)
+	if _, ok := config.LabelMap[action.TypeID]; ok { // checks if label can be found in the map, if so, it is stored in variable device
+		for _, instruction := range action.Message {
+			for _, comp := range config.LabelMap[action.TypeID] {
+				instruction.ComponentID = comp.ID
+				errorList = append(errorList,
+					checkActionDevice(Action{TypeID: comp.Device.ID, Message: []ComponentInstruction{instruction}}, config)...)
+			}
+		}
+	} else {
+		errorList = append(errorList,
+			fmt.Sprintf("label with id %s not found in map", action.TypeID))
+	}
+	return errorList
+}
+
+// checkValidComparison checks if the comparison is a valid one
+func checkValidComparison(comparison string) bool {
+	comparisonTypesAllowed := []string{"eq", "lt", "gt", "lte", "gte", "contains", "not"}
+	for _, comp := range comparisonTypesAllowed {
+		if comp == comparison {
+			return true
+		}
+	}
+	return false
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (or OrCondition) checkConstraints(config WorkingConfig, ruleID string) []string {
+	errorList := make([]string, 0)
+	for _, logic := range or.logics {
+		err := logic.checkConstraints(config, ruleID)
+		if err != nil {
+			errorList = append(errorList, err...)
+		}
+	}
+	return errorList
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (and AndCondition) checkConstraints(config WorkingConfig, ruleID string) []string {
+	errorList := make([]string, 0)
+	for _, logic := range and.logics {
+		err := logic.checkConstraints(config, ruleID)
+		if err != nil {
+			errorList = append(errorList, err...)
+		}
+	}
+	return errorList
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (condition Condition) checkConstraints(config WorkingConfig, ruleID string) []string {
```

Comment:
but the abstract method on top of the implementations, that reads more naturally